components/chip/src/chip/chip.js

```diff
@@ -0,0 +1,166 @@
+import { colors, theme } from '@dhis2/ui-constants'
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { Content } from './content.js'
+import { Icon } from './icon.js'
+import { Remove } from './remove.js'
+
+const DEFAULT_INLINE_MARGIN = '4'
+
+const Chip = ({
+    selected,
+    dense,
+    disabled,
+    dragging,
+    overflow,
+    className,
+    children,
+    onRemove,
+    onClick,
+    icon,
+    dataTest,
+    marginBottom,
+    marginLeft,
+    marginRight,
+    marginTop,
+    marginInlineStart,
+    marginInlineEnd,
+}) => {
+    const handleKeyDown = (event) => {
+        if (!onRemove) {
+            return
+        }
+        if (event.key === 'Backspace' || event.key === 'Delete') {
+            onRemove({}, event)
+        }
+    }
+
+    return (
+        <button
```

Comment:
Chips are mainly used for links. Button is definitely better than the previous span, but I think we should look into how we would support links, instead of having to do the navigation imperatively in `onClick` handler. It's not valid HTML to wrap a button in a link, so this component would have to support it. 

It might be worth supporting a `component`-prop, and `href` as props. See how material UI does it: https://mui.com/material-ui/react-chip/#clickable-link . A lot of times you would want to use a Chip with eg. `react-router`s `NavLink`. 

@kabaros @ismay

Reply:
it sounds correct .. but not in the scope of this PR, right? we can add a separate ticket for it

Reply:
You're right, but we may want to release that together. I brought it up because this PR changes the underlying element from span to a button. And wrapping a button in anchor tag is not valid, so it's potentially breaking .